src/Halogen/Driver.purs

```diff
@@ -131,7 +131,7 @@ runUI c s element = _.driver <$> do
         when (isJust rp) $ render ref
         putVar rpRef Nothing
         q
-      HaltHF -> empty
+      HaltHF msg -> throwError $ error msg
```

Comment:
We could probably do something to ensure we don't end up with hanging vars by doing something like this in the cases it might occur internally:

``` purescript
x <- thingThatMightFail
x <|> (putVar oldValue ref *> x) 
```